Compile failure on Ubuntu 22.04 with gcc 11.2

## Bug Report

### Greenplum version or build
master branch
### OS version and uname -a
Ubuntu 22.04
### autoconf options used ( config.status --config )
`./configure`
### Installation information ( pg_config )

### Expected behavior
Compile successfully.
### Actual behavior
```
gcc -Wall -Wmissing-prototypes -Wpointer-arith -Werror=vla -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -Wno-unused-but-set-variable -Werror=implicit-fallthrough=3 -Wno-format-truncation -Wno-stringop-truncation -O3  -Werror=uninitialized -Werror=implicit-function-declaration -I../../../../src/include   -D_GNU_SOURCE   -c -o gp_hyperloglog.o gp_hyperloglog.c
In function ‘gp_hll_compress_dense’,
    inlined from ‘gp_hll_compress’ at gp_hyperloglog.c:533:13:
gp_hyperloglog.c:584:11: error: ‘data’ may be used uninitialized [-Werror=maybe-uninitialized]
  584 |     len = pglz_compress(data,data_rawsize,dest,PGLZ_strategy_always);
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       ^~~~~~~~~~~~~

```
### Step to reproduce the behavior
`configure && make`

--- a/src/backend/utils/hyperloglog/gp_hyperloglog.c
+++ b/src/backend/utils/hyperloglog/gp_hyperloglog.c
@@ -571,6 +571,7 @@ gp_hll_compress_dense(GpHLLCounter hloglog)
                  errmsg("out of memory"),
                  errdetail("Failed on request of size %zu.", data_rawsize)));
     }
+    bzero(data, data_rawsize);
 
     /* put all registers in a normal array  i.e. remove dense packing so
      * lz compression can work optimally */

--- a/src/backend/utils/hyperloglog/gp_hyperloglog.c
+++ b/src/backend/utils/hyperloglog/gp_hyperloglog.c
@@ -571,7 +571,7 @@ gp_hll_compress_dense(GpHLLCounter hloglog)
                  errmsg("out of memory"),
                  errdetail("Failed on request of size %zu.", data_rawsize)));
     }
-    bzero(data, data_rawsize);
+    memset(data, 0, data_rawsize);
 
     /* put all registers in a normal array  i.e. remove dense packing so
      * lz compression can work optimally */

